crates/sui-types/src/object.rs

```diff
@@ -726,6 +726,45 @@ impl Object {
         )
     }
 
+    /// Create a system package which is not subject to size limits. Panics if the object ID is not
+    /// a known system package.
+    pub fn new_system_package(
+        modules: &[CompiledModule],
+        version: SequenceNumber,
+        dependencies: Vec<ObjectID>,
+        previous_transaction: TransactionDigest,
+    ) -> Self {
+        let ret = Self::new_package_from_data(
+            Data::Package(MovePackage::new_system(version, modules, dependencies)),
+            previous_transaction,
+        );
+
+        #[cfg(not(msim))]
+        assert!(ret.is_system_package());
+
+        ret
+    }
+}
+
+impl std::ops::Deref for Object {
+    type Target = ObjectInner;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::ops::DerefMut for Object {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        Arc::make_mut(&mut self.0)
+    }
+}
```

Comment:
When do we need this?

Reply:
Any time we assign through a `&mut Object` reference. It is used in temporary store when we write objects and update the latest transaction.

Reply:
I wonder if it's better to have an explicit swap API instead of &mut ref